src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1074,17 +1074,23 @@ public EnvVars getEnvironment() {
         if (buildData.getBuildsByBranchName().size() >= 100) {
             log.println("JENKINS-19022: warning: possible memory leak due to Git plugin usage; see: https://wiki.jenkins-ci.org/display/JENKINS/Remove+Git+Plugin+BuildsByBranch+BuildData");
         }
+        boolean checkForMultipleRevisions = false;
```

Comment:
The variable name is misleading. While it is handled correctly (```if (!checkForMultipleRevisions) { <do-catch-up-triggering> }```), the variable indicates that by default checks for multiple revisions / catch-up builds are disabled. I'd recommend changing the default value of this variable to true, negating the value from ```ext.disableMultipleRevisionDetection()``` in the for loop over all ```GitSCMExtensions``` and removing the negation from the if below.

Please make the default value for this ```true```.